src/service/data_products/taxa_count.py

```diff
@@ -234,9 +243,43 @@ async def get_taxa_counts(
         await _add_subset_data_in_place(q, store, collection_id, load_ver, rank, dp_proc)
         # For now always sort by the std data. See if ppl want sort by match/selection data
         # before implementing something more sophisticated.
+    _sort_taxa_counts(q)
+
     return _taxa_counts(dp_match=dp_match, dp_sel=dp_sel, data=q)
 
 
+def _sort_dict_list(
+        dict_list: list[dict],
+        key: str,
+        reverse: bool = True):
+    # Sort taxa count records, a list of dicts, in place by the key of dictionary with None values last.
+    # Key values should be either numeric or None.
+    # Certain keys, such as 'sel_count' or 'match_count,' may not exist in all dictionaries.
+    # For instance, these keys might be absent if the corresponding match or selection process did not occur/complete.
+    # In such cases, the default value of float('-inf') is used to ensure that the dictionaries are sorted correctly.
+    dict_list.sort(key=lambda x: (int(x.get(key)) if x.get(key) is not None else float('-inf')), reverse=reverse)
```

Comment:
Is this a stable sort?

Reply:
yea. `.sort` is stable. https://docs.python.org/3.11/howto/sorting.html#decorate-sort-undecorate

stable or not matters here?

Reply:
If you're doing multiple sorts and want the earlier sorts to be reflected in the final sort then the sort needs to be stable, yeah. Otherwise there's no guarantee that sort #1 won't be completely undone by sort #2

Reply:
I see. yea. `.sort` is stable.

Reply:
:+1: